cluster-autoscaler/provisioningrequest/checkcapacity/processor.go

```diff
@@ -77,13 +80,23 @@ func (p *checkCapacityProcessor) Process(provReqs []*provreqwrapper.Provisioning
 			break
 		}
 		conditions.AddOrUpdateCondition(provReq, v1beta1.BookingExpired, metav1.ConditionTrue, conditions.CapacityReservationTimeExpiredReason, conditions.CapacityReservationTimeExpiredMsg, metav1.NewTime(p.now()))
+		_, updErr := p.client.UpdateProvisioningRequest(provReq.ProvisioningRequest)
+		if updErr != nil {
+			klog.Errorf("failed to add BookingExpired condition to ProvReq %s/%s, err: %v", provReq.Namespace, provReq.Name, updErr)
+			continue
+		}
```

Comment:
What was the intention behind `maxUpdated`? If it was not to spend too much time doing the updates, maybe we should be incrementing `updated`  even if the call fails? Otherwise if e.g. all of the calls fail, maxUpdated is essentially ignored. Not sure how much of a problem this is